src/toil/batchSystems/slurm.py

```diff
@@ -68,9 +69,61 @@
     "STAGE_OUT",
     "STOPPED",
     "SUSPENDED"
-} 
+}
 
 class SlurmBatchSystem(AbstractGridEngineBatchSystem):
+    class PartitionInfo(NamedTuple):
+        partition_name: str
+        gres: bool
+        time: str
+        priority: str
+        cpus: str
+        memory: str
+
+    class PartitionSet:
+        """
+        Set of available partitions detected on the slurm batch system
+        """
+        default_gpu_partition: Optional[SlurmBatchSystem.PartitionInfo]
+        all_partitions: List[SlurmBatchSystem.PartitionInfo]
+        gpu_partitions: List[SlurmBatchSystem.PartitionInfo]
+
+        def __init__(self) -> None:
+            self._get_partition_info()
+            self._get_gpu_partitions()
+
+        def _get_gpu_partitions(self) -> None:
+            """
+            Get all available GPU partitions. Also get the default GPU partition.
+            :return: None
+            """
+            self.gpu_partitions = [partition for partition in self.all_partitions if partition.gres]
+            # Grab the lowest priority GPU partition
+            # If no GPU partitions are available, then set the default to None
+            self.default_gpu_partition = None
+            if len(self.gpu_partitions) > 0:
+                self.default_gpu_partition = sorted(self.gpu_partitions, key=lambda x: x.priority)[0]
```

Comment:
If we want to make the GPU partitions a set of names we need to change this.

```suggestion
            gpu_partitions = [partition for partition in self.all_partitions if partition.gres]
            self.gpu_partitions = {p.name for p in gpu_partitions}
            # Grab the lowest priority GPU partition
            # If no GPU partitions are available, then set the default to None
            self.default_gpu_partition = None
            if len(gpu_partitions) > 0:
                self.default_gpu_partition = sorted(gpu_partitions, key=lambda x: x.priority)[0]
```